--enable-parallel flag prevents libcuba.a to be found

Given that I manually compiled and installed Cuba, I'm experiencing the following issue on OSX:
```
$ ./configure --with-cuba --enable-parallel
[...]
checking for cuba.h in system default directories
checking cuba.h usability... yes
checking cuba.h presence... yes
checking for cuba.h... yes
checking for cuba library in system default directories
detected cuba matching the interface of version 4.1
checking for main in -lcuba... no
configure: error: Could not find libcuba.a; use --with-cuba-lib-dir=DIR or compile without cuba
```
```
$ ./configure --with-cuba
[...]
checking for cuba.h in system default directories
checking cuba.h usability... yes
checking cuba.h presence... yes
checking for cuba.h... yes
checking for cuba library in system default directories
detected cuba matching the interface of version 4.1
checking for main in -lcuba... yes
[...]
```
I also tried to manually specify the path to `libcuba.a`, no luck. Can you reproduce the issue?

--- a/configure.ac
+++ b/configure.ac
@@ -49,17 +49,26 @@ AC_ARG_ENABLE(
   [AS_HELP_STRING([--enable-parallel],[compile with OpenMP thread parallelization (default no)])],
   [
     case "${enableval}" in
-      yes) parallelization=1 ;
-        CXXFLAGS+=" -fopenmp" ;
-        LDFLAGS+=" -fopenmp" ;
-        DEP_CFLAGS="$DEP_CFLAGS -fopenmp" ;
-        DEP_LIBS="$DEP_LIBS -fopenmp" ;;
+      yes) parallelization=1 ;;
       no)  parallelization=0 ;;
       *) AC_MSG_ERROR([bad value ${enableval} for --enable-parallel]) ;;
     esac
   ],
   [parallelization=0]
 )
+
+m4_include(tools/build/ax_openmp.m4)
+if test $parallelization = 1; then
+  AX_OPENMP(
+    [
+    CXXFLAGS+=" $OPENMP_CXXFLAGS";
+    LDFLAGS+=" $OPENMP_CXXFLAGS";
+    DEP_CFLAGS+=" $OPENMP_CXXFLAGS";
+    DEP_LIBS+=" $OPENMP_CXXFLAGS";
+    ],
+    [AC_MSG_ERROR([Cannot enable parallelization without OpenMP])]
+)
+fi
 
 AC_DEFINE_UNQUOTED([THREAD_PARALLELIZATION], [$parallelization], [OpenMP thread parallelization])
 AM_CONDITIONAL([THREAD_PARALLELIZATION], [test $parallelization = 1])
@@ -320,7 +329,7 @@ ROOT_PATH(
   ,
   [
     AC_MSG_ERROR([ROOT cannot be found. Please ensure that it is installed
-      and that either \$ROOTSYS is defined corectly or root-config is in the path.
+      and that either \$ROOTSYS is defined correctly or root-config is in the path.
       One can pass --with-rootsys=/path/to/rootsys to configure as well.])
     exit
   ]

--- a/tools/build/ax_openmp.m4
+++ b/tools/build/ax_openmp.m4
@@ -1,0 +1,123 @@
+# ===========================================================================
+#        https://www.gnu.org/software/autoconf-archive/ax_openmp.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_OPENMP([ACTION-IF-FOUND[, ACTION-IF-NOT-FOUND]])
+#
+# DESCRIPTION
+#
+#   This macro tries to find out how to compile programs that use OpenMP a
+#   standard API and set of compiler directives for parallel programming
+#   (see http://www-unix.mcs/)
+#
+#   On success, it sets the OPENMP_CFLAGS/OPENMP_CXXFLAGS/OPENMP_F77FLAGS
+#   output variable to the flag (e.g. -omp) used both to compile *and* link
+#   OpenMP programs in the current language.
+#
+#   NOTE: You are assumed to not only compile your program with these flags,
+#   but also link it with them as well.
+#
+#   If you want to compile everything with OpenMP, you should set:
+#
+#     CFLAGS="$CFLAGS $OPENMP_CFLAGS"
+#     #OR#  CXXFLAGS="$CXXFLAGS $OPENMP_CXXFLAGS"
+#     #OR#  FFLAGS="$FFLAGS $OPENMP_FFLAGS"
+#
+#   (depending on the selected language).
+#
+#   The user can override the default choice by setting the corresponding
+#   environment variable (e.g. OPENMP_CFLAGS).
+#
+#   ACTION-IF-FOUND is a list of shell commands to run if an OpenMP flag is
+#   found, and ACTION-IF-NOT-FOUND is a list of commands to run it if it is
+#   not found. If ACTION-IF-FOUND is not specified, the default action will
+#   define HAVE_OPENMP.
+#
+# LICENSE
+#
+#   Copyright (c) 2008 Steven G. Johnson <stevenj@alum.mit.edu>
+#   Copyright (c) 2015 John W. Peterson <jwpeterson@gmail.com>
+#   Copyright (c) 2016 Nick R. Papior <nickpapior@gmail.com>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
+#   gives unlimited permission to copy, distribute and modify the configure
+#   scripts that are the output of Autoconf when processing the Macro. You
+#   need not follow the terms of the GNU General Public License when using
+#   or distributing such scripts, even though portions of the text of the
+#   Macro appear in them. The GNU General Public License (GPL) does govern
+#   all other use of the material that constitutes the Autoconf Macro.
+#
+#   This special exception to the GPL applies to versions of the Autoconf
+#   Macro released by the Autoconf Archive. When you make and distribute a
+#   modified version of the Autoconf Macro, you may extend this special
+#   exception to the GPL to apply to your modified version as well.
+
+#serial 13
+
+AC_DEFUN([AX_OPENMP], [
+AC_PREREQ([2.69]) dnl for _AC_LANG_PREFIX
+
+AC_CACHE_CHECK([for OpenMP flag of _AC_LANG compiler], ax_cv_[]_AC_LANG_ABBREV[]_openmp, [save[]_AC_LANG_PREFIX[]FLAGS=$[]_AC_LANG_PREFIX[]FLAGS
+ax_cv_[]_AC_LANG_ABBREV[]_openmp=unknown
+# Flags to try:  -fopenmp (gcc), -mp (SGI & PGI),
+#                -qopenmp (icc>=15), -openmp (icc),
+#                -xopenmp (Sun), -omp (Tru64),
+#                -qsmp=omp (AIX),
+#                none
+ax_openmp_flags="-fopenmp -openmp -qopenmp -mp -xopenmp -omp -qsmp=omp none"
+if test "x$OPENMP_[]_AC_LANG_PREFIX[]FLAGS" != x; then
+  ax_openmp_flags="$OPENMP_[]_AC_LANG_PREFIX[]FLAGS $ax_openmp_flags"
+fi
+for ax_openmp_flag in $ax_openmp_flags; do
+  case $ax_openmp_flag in
+    none) []_AC_LANG_PREFIX[]FLAGS=$save[]_AC_LANG_PREFIX[] ;;
+    *) []_AC_LANG_PREFIX[]FLAGS="$save[]_AC_LANG_PREFIX[]FLAGS $ax_openmp_flag" ;;
+  esac
+  AC_LINK_IFELSE([AC_LANG_SOURCE([[
+@%:@include <omp.h>
+
+static void
+parallel_fill(int * data, int n)
+{
+  int i;
+@%:@pragma omp parallel for
+  for (i = 0; i < n; ++i)
+    data[i] = i;
+}
+
+int
+main()
+{
+  int arr[100000];
+  omp_set_num_threads(2);
+  parallel_fill(arr, 100000);
+  return 0;
+}
+]])],[ax_cv_[]_AC_LANG_ABBREV[]_openmp=$ax_openmp_flag; break],[])
+done
+[]_AC_LANG_PREFIX[]FLAGS=$save[]_AC_LANG_PREFIX[]FLAGS
+])
+if test "x$ax_cv_[]_AC_LANG_ABBREV[]_openmp" = "xunknown"; then
+  m4_default([$2],:)
+else
+  if test "x$ax_cv_[]_AC_LANG_ABBREV[]_openmp" != "xnone"; then
+    OPENMP_[]_AC_LANG_PREFIX[]FLAGS=$ax_cv_[]_AC_LANG_ABBREV[]_openmp
+  fi
+  m4_default([$1], [AC_DEFINE(HAVE_OPENMP,1,[Define if OpenMP is enabled])])
+fi
+])dnl AX_OPENMP

